### ISSUE TITLE
MSRV build is broken

### ISSUE THREAD

**Issue by franziskuskiefer** [bheisler/criterion.rs#614]
Running

```
cargo +nightly update -Z minimal-versions
cargo build
```

fails with

```
error[E0433]: failed to resolve: use of undeclared crate or module `syntax`
 --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-1.5.0/src/literal/mod.rs:9:9
  |
9 |     use syntax::hir::literal::Literals;
  |         ^^^^^^ use of undeclared crate or module `syntax`
```

This blocks updating criterion in https://github.com/RustCrypto/formats/pull/720.

**Comment by elpiel** [comment]
MSRV should be made > 1.59 based on the latest `master` branch

> error: package `async-global-executor v2.3.0` cannot be built because it requires rustc 1.59 or newer, while the currently active rustc version is 1.57.0

https://github.com/bheisler/criterion.rs/actions/runs/3028960427/jobs/4874098488

**Comment by dhardy** [comment]
https://github.com/rust-random/rand/pull/1329 is currently blocked on our MSRV (1.60 in rand master branch).

Testing now wants at least 1.70:
```
error: package `clap_lex v0.6.0` cannot be built because it requires rustc 1.70.0 or newer, while the currently active rustc version is 1.60.0
```

Does Criterion have any particular policy here on the MSRV?

**Comment by elpiel** [comment]
@dhardy you can check - https://github.com/bheisler/criterion.rs#compatibility-policy

I believe msrv policy is less strict mainly because criterion is used primarily in dev-dependencies

**Comment by MarijnS95** [comment]
Notice that this is not an "MSRV build" but a `-Zminimal-versions` build. It's mentioned in https://github.com/rust-lang/regex/issues/931 as a wontfix (or rather, won't yank 😞) in `regex`, and was recently fixed in `criterion` by bumping to `regex 1.5.1` in https://github.com/bheisler/criterion.rs/pull/821. This can probably be closed.
